internal/sys/fs_test.go

```diff
@@ -274,3 +274,89 @@ func TestFSContext_Renumber(t *testing.T) {
 		require.Equal(t, syscall.ENOTSUP, fsc.Renumber(3, 3))
 	})
 }
+
+func TestReaddDir_Rewind(t *testing.T) {
+	tests := []struct {
+		name           string
+		f              *Readdir
+		cookie         int64
+		expectedCookie int64
+		expectedErrno  syscall.Errno
+	}{
+		{
+			name: "no prior call",
+		},
+		{
+			name:          "no prior call, but passed a cookie",
+			cookie:        1,
+			expectedErrno: syscall.EINVAL,
+		},
+		{
+			name: "cookie is negative",
+			f: &Readdir{
+				cursor: 3,
+			},
+			cookie:        -1,
+			expectedErrno: syscall.EINVAL,
+		},
+		{
+			name: "cookie is greater than last d_next",
+			f: &Readdir{
+				cursor: 3,
+			},
+			cookie:        5,
+			expectedErrno: syscall.EINVAL,
+		},
+		{
+			name: "cookie is last pos",
+			f: &Readdir{
+				cursor: 3,
+			},
+			cookie: 3,
+		},
+		{
+			name: "cookie is one before last pos",
+			f: &Readdir{
+				cursor: 3,
+			},
+			cookie: 2,
+		},
+		// fixme? we are no longer using a sliding window
+		//{
+		//	name: "cookie is before current entries",
+		//	f: &Readdir{
+		//		countRead: 5,
+		//		//dirents:   testDirents,
+		//	},
+		//	cookie:        1,
+		//	expectedErrno: syscall.ENOSYS, // not implemented
+		//},
```

Comment:
I have kept this test here because it would not have been clear from the diff that it was deleted. I will delete it if we're ok with this: basically, it makes no longer sense because we are no longer using a sliding window. Should we go back to a sliding window instead?

Reply:
If I'm misreading, my mistake. a bit jet lagged.

First, I'd like to see what this test behavior is with the current change as it is hard to guess.

The sliding window was there to avoid caching a potentially massive directory listing, yet still allow lookup of recent entries. It is indeed simpler to not do a sliding window, but yeah that re-introduces this problem which we fixed earlier.

If the main issue is about diff size, maybe there's a way to extract some of this PR into another change, retaining the windowing. OTOH I'm also fine to have this wait until windowing is here. What we shouldn't do is merge partial implementation of something already supported as we should be able to ship main whenever we need to. I hope this makes sense.

Reply:
It returns ESUCCESS instead of ENOSYS. The test was trying to seek to a position before the window by sending a cookie with a lower value.

Reply:
restored.